src/ProgrammableDevice.cc

```diff
@@ -0,0 +1,109 @@
+#include "ProgrammableDevice.hh"
+
+#include "CommandController.hh"
+#include "MSXCliComm.hh"
+#include "MSXCPUInterface.hh"
+#include "MSXException.hh"
+#include "StringSetting.hh"
+#include "serialize.hh"
+
+#include "narrow.hh"
+
+namespace openmsx {
+
+ProgrammableDevice::ProgrammableDevice(const DeviceConfig& config)
+	: MSXDevice(config)
+	, portListSetting(
+		getCommandController(), tmpStrCat(getName(), " ports"),
+		"List of ports are added to the pool of IO ports this extension scans for", "")
+	, resetCallback(getCommandController(),
+		getName() + " reset callback",
+		"Tcl proc to call when reset was detected, no parameters and return values are "
+		"used", "", Setting::Save::NO)
+	, readCallback(getCommandController(),
+		getName() + " read callback",
+		"Defines a Tcl proc that receives a port number (byte) and returns a byte when "
+		"the MSX is reading a port from the device", "", Setting::Save::NO)
+	, writeCallback(getCommandController(),
+		getName() + " write callback",
+		"Defines a Tcl proc that receives a port number (byte) and a byte when the "
+		"MSX is writing a port from the device", "", Setting::Save::NO)
+{
+	portListSetting.attach(*this);
+	try {
+		outPorts = inPorts = parseAllPorts(portListSetting);
+	} catch (MSXException&) {
+		// silently ignore error in settings.xml
+		portListSetting.setString("");
+	}
+}
+
+ProgrammableDevice::~ProgrammableDevice()
+{
+	portListSetting.detach(*this);
+}
+
+IterableBitSet<256> ProgrammableDevice::parseAllPorts(const Setting& setting)
+{
+	Interpreter& interp = getCommandController().getInterpreter();
+	TclObject portStr(((StringSetting&)setting).getString());
+	IterableBitSet<256> ports = {};
```

Comment:
This is not wrong. And in some coding styles this is even recommended. But in openMSX we'd usually write this just like:
```
IterableBitSet<256> ports;
```
No need to call the default constructor (more) explicitly.